bundles/org.openhab.binding.nanoleaf/src/main/java/org/openhab/binding/nanoleaf/internal/colors/NanoleafPanelColors.java

```diff
@@ -0,0 +1,148 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.nanoleaf.internal.colors;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.nanoleaf.internal.handler.NanoleafPanelHandler;
+import org.openhab.core.library.types.HSBType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Stores information about panels and their colors, while sending notifications to panels and controllers
+ * about updated states.
+ *
+ * @author Jørgen Austvik - Refactoring to own class
+ */
+@NonNullByDefault
+public class NanoleafPanelColors {
+
+    private final Logger logger = LoggerFactory.getLogger(NanoleafPanelColors.class);
+
+    // holds current color data per panel
+    private final Map<Integer, HSBType> panelColors = new ConcurrentHashMap<>();
+    private final Map<Integer, NanoleafPanelColorChangeListener> panelChangeListeners = new ConcurrentHashMap<>();
+    private final List<NanoleafControllerColorChangeListener> controllerListeners = new CopyOnWriteArrayList<>();
+
+    private boolean updatePanelColorNoController(Integer panelId, HSBType color) {
```

Comment:
Here and in other method signatures in this class: Could primitive type (`int`) be used?

Reply:
It could, but the map on the inside has to have Integer, so this saves some boxing in case the variable on the outside is an Integer.